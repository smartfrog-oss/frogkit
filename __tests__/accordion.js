import { mount, shallow } from 'vue-test-utils'
import Accordion from '@/components/accordion'

const propsData = {
  placeholder: 'Title',
  open: false
}

describe('Accordion component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = mount(Accordion, { propsData })
    expect(wrapper.html()).toMatchSnapshot()
  })


  it('Should open and close', () => {
    const cmp = shallow(Accordion, { propsData })
    expect(cmp.vm.isOpen).toBe(false)
    cmp.vm.toggle()
    expect(cmp.vm.isOpen).toBe(true)
    cmp.vm.toggle()
    expect(cmp.vm.isOpen).toBe(false)
  })

  it('Should close and open', () => {
    const cmp = shallow(Accordion, { propsData: {...propsData, open: true} })
    expect(cmp.vm.isOpen).toBe(true)
    cmp.vm.toggle()
    expect(cmp.vm.isOpen).toBe(false)
    cmp.vm.toggle()
    expect(cmp.vm.isOpen).toBe(true)
  })

})
