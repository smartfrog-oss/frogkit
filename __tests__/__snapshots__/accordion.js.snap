// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Should render component and match snapshot 1`] = `
<section class="fk-accordion">
  <flex justify="space-between" align="center" class="fk-accordion__header">
    <title size="xs" class="fk-accordion__title"> Title </title>
    <icon icon="plus" class="fk-accordion__icon"></icon>
  </flex>
  <!---->
</section>
`;
